sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -353,7 +371,109 @@ struct graphs_support;
 } // namespace device
 } // namespace info
 
-class node {};
+enum class dynamic_parameter_type {
+  usm,
+  accessor,
+  scalar,
+  raw
+};
+
+template <dynamic_parameter_type DynParamType>
+class dynamic_parameter;
+
+template <>
+class dynamic_parameter<dynamic_parameter_type::usm> {
+public:
+  dynamic_parameter(command_graph<graph_state::modifiable> graph,
+                    std::string label = "");
+
+  void register(void* param, node node);
+
+  void register(void* param);
+
+  void register(int argIndex, node node);
+
+  void update(void* newValue,
+              command_graph<graph_state::executable> execGraph);
+
+  const std::string& get_label() const;
+};
+
+template <>
+class dynamic_parameter<dynamic_parameter_type::accessor> {
+public:
+  dynamic_parameter(command_graph<graph_state::modifiable> graph,
+                    std::string label = "");
+
+  template <typename T>
+  void register(T param, node node);
+
+  template <typename T>
+  void register(T param);
+
+  void register(int argIndex, node node);
+
+  template <typename T>
+  void update(T newValue,
+              command_graph<graph_state::executable> execGraph);
+
+  const std::string& get_label() const;
+};
+
+
+template <>
+class dynamic_parameter<dynamic_parameter_type::scalar> {
+public:
+  dynamic_parameter(command_graph<graph_state::modifiable> graph,
+                    size_t size, std::string label = "");
+
+  template <typename T>
+  void register(T param, node node);
+
+  template <typename T>
+  void register(T param);
+
```

Comment:
I think these cases are problematic due to potential collisions when trying to find the parameter inside the node. For example, if there are multiple parameters with the same type (though we can't actually compare types directly anyway due to the loss of type information in the node), size and memory contents, which one is the intended target for registration and update?

The case where we know the `argIndex` is fine because it is explicit, but since scalar parameters to kernels in SYCL are already quite hidden due to the nature of lambda capture, I think these methods would lead to some very unintuitive behaviour that would require a lot of documentation about avoiding collisions to try and mitigate. I believe they should be removed for now unless there is a strong desire/use case for this right now, and could be revisited in the future if such a need arises.

I believe this also applies to the raw type below which I think should also be restricted to the case of explicit indices only.

Reply:
Agree with this being likely to cause collisions, without it I guess we won't be able to support the use case of a library not having access to the individual `node` handles and wanting to update scalar/raw arguments. I don't think this is unreasonable however, and can consider reintroducing the API again if required.

@guoyejun Do you know if you will need to update scalars in your usage where you made the requirement of not having access to node handles?

Reply:
@EwanC  I currently do not see the requirement to update scalars and raw type, agree to consider reintroducing the API again when required.